apollo-router/src/plugins/authentication/mod.rs

```diff
@@ -532,11 +539,13 @@ fn authenticate(
 
     // Make sure the format of our message matches our expectations
     // Technically, the spec is case sensitive, but let's accept
-    // case variations
-    //
+    // case variations. Furthermore, if the user has configured to ignore
```

Comment:
It would be simpler as:
```
if jwt_value.len() < prefix_len
            || !&jwt_value[..prefix_len].eq_ignore_ascii_case(&config.header_value_prefix {
    if config.ignore_other_prefixes {
        return ControlFlow::Continue(request);
    } else {
        return failure_message(
            request.context,
        );
    }
}
```